src/Path/UploadListings.jsx

```diff
@@ -104,47 +102,108 @@ function UploadListings() {
 
   function handleInputChange(e) {
     e.preventDefault();
-    const file = e.target.files[0];
-    if (file) {
-      const MAX_IMAGE_SIZE_MB = 20;
+    const files = e.target.files;
+    const MAX_IMAGE_SIZE_MB = 20;
+    let hasImage = false;
+    let hasPdf = false;
+
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i];
+
       if (file.size > MAX_IMAGE_SIZE_MB * 1024 * 1024) {
         alert(`Maximum file size is ${MAX_IMAGE_SIZE_MB} MB`);
         return;
       }
 
       if (file.type.startsWith("image/")) {
-        setLocalImageOrPdf(true);
-        setImage(e.target.files);
+        hasImage = true;
       } else if (file.type === "application/pdf") {
-        setLocalImageOrPdf(true);
-        setPdf(file);
-        setListingInput((prev) => ({
-          ...prev,
-          hasAttachment: true,
-        }));
+        hasPdf = true;
       }
     }
+
+    if (hasImage && hasPdf) {
+      alert(t("imagePdfAlert"));
+      return;
+    }
+
+    if (hasImage) {
+      setLocalImageOrPdf(true);
+      setImage(files);
+    }
+
+    if (hasPdf) {
+      setLocalImageOrPdf(true);
+      setPdf(files[0]); // Assuming only one PDF file is allowed
+      setListingInput((prev) => ({
+        ...prev,
+        hasAttachment: true,
+      }));
+    }
   }
 
   const [localImages, setLocalImages] = useState([]);
   const handleMultipleInputChange = (event) => {
-    const newImages = Array.from(event.target.files);
-    setLocalImages((prevImages) => [...prevImages, ...newImages]);
-    setImage([...image, ...newImages]);
+    const newFiles = Array.from(event.target.files);
+
+    if (image.length > 0) {
+      const validImages = newFiles.filter(file => file.type.startsWith("image/"));
+      const invalidFiles = newFiles.filter(file => !file.type.startsWith("image/") && file.type !== "application/pdf");
```

Comment:
the and condition is not making sense. You can remove that and just keep newFiles.filter(file => !file.type.startsWith("image/"))

Reply:
do the same below